observation/beamformer_raster_scan.py

```diff
@@ -0,0 +1,206 @@
+#!/usr/bin/env python
+#
+# Perform raster scan using tied-array beam.
+#
+# Ludwig Schwardt
+# 11 August 2021
+#
+
+import time
+
+import numpy as np
+
+from katcorelib.observe import (
+    SessionCBF,
+    collect_targets,
+    standard_script_options,
+    start_session,
+    user_logger,
+    verify_and_connect,
+)
+from katcorelib.targets import Offset
+
+
+def set_beamformer_gains(cbf, ants, gain):
+    """"""
+    for stream in cbf.beamformers:
+        if not stream.name.endswith('0x'):
+            continue
+        # Set beamformer quantization gain.
+        bq_gain = gain / np.sqrt(len(ants))
+        reply = stream.req.quant_gains(bq_gain)
+        if reply.succeeded:
+            user_logger.info('Beamformer stream %s quantization gain set to '
+                             '%g for %d antennas.', stream, bq_gain, len(ants))
+        else:
+            user_logger.warning('Failed to set beamformer quantization gain %g on '
+                                'stream %s!', bq_gain, stream)
+
+
+def set_beamformer_weights(cbf, ants):
+    """"""
+    for stream in cbf.beamformers:
+        if not stream.name.endswith('0x'):
+            continue
+        # Set the beamformer weights (TAB shaping).
+        user_logger.info('Setting beamformer weights for stream %s:', stream)
+        weights = []
+        for inp in stream.inputs:
+            weight = 1.0 if inp[:-1] in ants else 0.0
+            weights.append(weight)
+            user_logger.info('  input %r will get weight %f', inp, weight)
+        reply = stream.req.weights(*weights)
+        if reply.succeeded:
+            user_logger.info('Set input weights successfully')
+        else:
+            user_logger.warning('Failed to set input weights!')
+
+
+def point(session, target, duration):
+    """"""
+    user_logger.info('  pointing at %s for %g seconds', target, duration)
+    before = time.time()
+    for stream in session.cbf.beamformers:
+        if not stream.name.endswith('0x'):
+            continue
+        session.cbf.req.set_beam_target('wide.tied-array-channelised-voltage.0x', target)
+        # user_logger.info('  (%s -> %s)', stream.name, target)
+    time_spent = time.time() - before
+    time.sleep(duration - time_spent)
```

Comment:
Is the use of this script likely to have  time_spent ~ pointing_duration .  There would be no actual dwell time in this case